### PR TITLE
Update changePassword Swagger and add updateAccount

### PR DESCRIPTION
**Pull request by minhdev313**
Nhờ Minh review qua nhé

### CODE REVIEW

File: internal/controller/auth.controller.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ func (ac *AuthController) ResetPassword(ctx *gin.Context) {
 // @Accept json
 // @Produce json
 // @Param data body auth_dto.ChangePasswordInput true "data"
-// @Router /reset-password [put]
+// @Router /change-password [put]
```
Comment (minhbear) [inline internal/controller/auth.controller.go:180]: Did you change the path in auth router ?

File: internal/service/admin/admin.service.go [hunk 2]
```diff
@@ -16,13 +16,15 @@ import (
...
 	"go.uber.org/zap"
+	"gorm.io/gorm"
 )
 
 type IAdminService interface {
 	CreateStudentAccount(ctx *gin.Context, input *admin_dto.AdminCreateStudentAccountInput) (int, error)
 	CreateTeacherAccount(ctx *gin.Context, input *admin_dto.AdminCreateTeacherAccountInput) (int, error)
 	UploadFileStudentData(ctx *gin.Context, file *multipart.FileHeader) (int, *import_dto.ImportOutput)
 	UploadFileTeacherData(ctx *gin.Context, file *multipart.FileHeader) (int, *import_dto.ImportOutput)
+	UpdateAccount(ctx *gin.Context, userID int64, input *admin_dto.UpdateAccountInput) (int, error)
```
Comment (minhbear) [inline internal/service/admin/admin.service.go]: Pass the userID into UpdateAccountInput, the tag will be swagger ignore and add the value to the input in the controller instead pass 3 params like this

File: internal/service/admin/admin.service.go [hunk 3]
```diff
@@ -188,3 +190,100 @@ func (as *adminService) createAccount(ctx *gin.Context, input InputCreateAccount
...
+		message := global.Localizer.MustLocalize(&i18n.LocalizeConfig{
+			MessageID: constant.MessageI18nId.UpdateFailed,
+		})
+		return http.StatusInternalServerError, errors.New(message)
+	}
+
+	switch user.UserType {
+	case constant.UserType.Student:
+		var student model.Student
+		if tx.First(&student, "user_id = ?", user.ID).Error == nil {
```
Comment (minhbear) [inline internal/service/admin/admin.service.go]: Need to check if error != nil first, in this case return Internal server error and add the log not found student and then do same step above

File: internal/service/admin/admin.service.go [hunk 4]
```diff
@@ -188,3 +190,100 @@ func (as *adminService) createAccount(ctx *gin.Context, input InputCreateAccount
...
+				tx.Rollback()
+				message := global.Localizer.MustLocalize(&i18n.LocalizeConfig{
+					MessageID: constant.MessageI18nId.UpdateStudentDetailsFailed,
+				})
+				return http.StatusInternalServerError, errors.New(message)
+			}
+		}
+	case constant.UserType.Teacher:
+		var teacher model.Teacher
+		if tx.First(&teacher, "user_id = ?", user.ID).Error == nil {
```
Comment (minhbear) [inline internal/service/admin/admin.service.go]: same above

File: internal/controller/auth.controller.go (hunk 1 above)
Comment (minhdev313) [reply internal/controller/auth.controller.go:180]: Yes i did. I checked on Swagger UI and recognized it's wrong so i fixed that.
image

**Pull request merged**